pay-queue/src/pay_queue/external/gcp_auth.py

```diff
@@ -13,28 +13,33 @@
 
 
 def verify_jwt(session):
-    """Verify token is valid."""
-    msg = ''
+    """Verify token is valid and check email claims."""
     try:
         # Get the Cloud Pub/Sub-generated JWT in the "Authorization" header.
-        id_token.verify_oauth2_token(
-            request.headers.get('Authorization').split()[1],
+        jwt_token = request.headers.get('Authorization', '').split()[1]
+        # Verify the JWT and get the claims
+        claim = id_token.verify_oauth2_token(
+            jwt_token,
             Request(session=session),
             audience=current_app.config.get('PAY_SUB_AUDIENCE')
         )
-    except Exception as e:  # TODO fix
-        msg = f'Invalid token: {e}\n'
-    finally:
-        return msg
+        # Check if the email is verified and matches the configured email
+        required_email = current_app.config.get('VERIFY_PUBSUB_EMAIL')
+        if not claim.get('email_verified') or claim.get('email') != required_email:
+            return 'Email not verified or does not match', 400
+    except Exception as e:
+        return f'Invalid token: {e}', 400
+    return None
 
 
 def ensure_authorized_queue_user(f):
     """Ensures the user is authorized to use the queue."""
     @functools.wraps(f)
     def decorated_function(*args, **kwargs):
         # Use CacheControl to avoid re-fetching certificates for every request.
-        if message := verify_jwt(CacheControl(Session())):
-            print(message)
-            abort(HTTPStatus.UNAUTHORIZED)
+        if current_app.config.get('VERIFY_PUBSUB_VIA_JWT', False):
+            if message := verify_jwt(CacheControl(Session())):
+                print(message)
```

Comment:
print(message) -> current_app.logging?

Reply:
I'll remove it, this is a for my local testing, I forgot deleting it